src/node/byzantine_identity.h

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/tx_id.h"
+#include "consensus/aft/impl/state.h"
+#include "crypto/pem.h"
+#include "crypto/verifier.h"
+#include "enclave/rpc_sessions.h"
+#include "kv/kv_types.h"
+#include "node/identity.h"
+#include "node/rpc/call_types.h"
+#include "node/rpc/serdes.h"
+#include "node/rpc/serialization.h"
+#include "service_map.h"
+
+#include <vector>
+
+namespace ccf
+{
+  struct ByzantineIdentity : public Identity
+  {
+    ByzantineIdentity() : Identity()
+    {
+      type = IdentityType::BYZANTINE;
+    }
+  };
+
+  DECLARE_JSON_TYPE(ByzantineIdentity);
+  DECLARE_JSON_REQUIRED_FIELDS(ByzantineIdentity, cert);
+
+  using ByzantineIdentities =
+    ServiceMap<kv::ReconfigurationId, ByzantineIdentity>;
+
+  class ByzantineIdentitiesHook : public kv::ConsensusHook
+  {
+    kv::Version version;
+    std::unordered_map<kv::ReconfigurationId, ByzantineIdentity> byids;
+
+  public:
+    ByzantineIdentitiesHook(
+      kv::Version version_, const ByzantineIdentities::Write& w) :
+      version(version_)
+    {
+      for (const auto& [rid, opt_bid] : w)
+      {
+        if (opt_bid.has_value())
+        {
+          LOG_DEBUG_FMT(
+            "New Byzantine network identity, valid for configuration #{}.",
+            rid);
+          byids.try_emplace(rid, opt_bid.value());
+        }
+      }
+    }
+
+    void call(kv::ConfigurableConsensus* consensus) override
+    {
+      for (auto& [rid, id] : byids)
+      {
+        consensus->add_identity(version, rid, id);
+      }
+    }
+  };
+
+  class ByzantineIdentityTracker
```

Comment:
Is is possible to unit test this class?

Reply:
It really doesn't do anything testable yet, it just keeps a list of configuration IDs. This is where the actual Byzantine identity crypto will go, which will of course come with tests.